### PR TITLE
DB-9330 Add option for batched newTimestamp RPCs

### PR DESCRIPTION
**Pull request by dgomezferro**
This PR also includes: DB-7371 Port Timestamp Client/Server to Netty 4

### CODE REVIEW

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1611/

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by hatyo — APPROVED** [review]
Great work!

File: splice_timestamp_api/src/main/java/com/splicemachine/timestamp/impl/TimestampConnection.java [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+     */
+    protected int getPort() {
+        return timestampHostProvider.getPort();
+    }
+
+    protected void connectIfNeeded() throws TimestampIOException{
+
+        // Even though state is an atomic reference, synchronize on whole block
+        // including code that attempts connection. Otherwise, two threads might
+        // end up trying to connect at the same time.
```
Comment (msirek) [inline splice_timestamp_api/src/main/java/com/splicemachine/timestamp/impl/TimestampConnection.java:164]: Is this comment still relevant?  I don't see the synchronize block.

**Review by msirek — APPROVED** [review]
+1

**Review by carolp-503 — APPROVED** [review]
+1

File: splice_timestamp_api/src/main/java/com/splicemachine/timestamp/impl/TimestampConnection.java (hunk 1 above)
Comment (dgomezferro) [reply splice_timestamp_api/src/main/java/com/splicemachine/timestamp/impl/TimestampConnection.java:164]: It looks outdated, I moved that directly from TimestampClient, if I get a chance I'll update it. Thanks!

**Review by jyuanca — APPROVED** [review]

**Pull request merged**